phidgets_spatial/src/spatial_ros_i.cpp

```diff
@@ -0,0 +1,295 @@
+#include <functional>
+#include <memory>
+#include <mutex>
+
+#include <ros/ros.h>
+#include <ros/service_server.h>
+#include <sensor_msgs/Imu.h>
+#include <sensor_msgs/MagneticField.h>
+#include <std_msgs/Bool.h>
+#include <std_srvs/Empty.h>
+
+#include "phidgets_api/spatial.h"
+#include "phidgets_spatial/spatial_ros_i.h"
+
+namespace phidgets {
+
+SpatialRosI::SpatialRosI(ros::NodeHandle nh, ros::NodeHandle nh_private)
+    : nh_(nh), nh_private_(nh_private)
+{
+    ROS_INFO("Starting Phidgets SPATIAL");
+
+    ROS_INFO("Opening spatial");
+    int serial_num;
+    if (!nh_private_.getParam("serial", serial_num))
+    {
+        serial_num = -1;  // default open any device
+    }
+    int hub_port;
+    if (!nh_private.getParam("hub_port", hub_port))
+    {
+        hub_port = 0;  // only used if the device is on a VINT hub_port
+    }
+    if (!nh_private_.getParam("frame_id", frame_id_))
+    {
+        // As specified in http://www.ros.org/reps/rep-0145.html
+        frame_id_ = "imu_link";
+    }
+    if (!nh_private_.getParam("linear_acceleration_stdev",
+                              linear_acceleration_stdev_))
+    {
+        linear_acceleration_stdev_ = 300.0 * 1e-6 * G;  // 300 ug as per manual
+    }
+    if (!nh_private_.getParam("angular_velocity_stdev",
+                              angular_velocity_stdev_))
+    {
+        angular_velocity_stdev_ =
+            0.02 * (M_PI / 180.0);  // 0.02 deg/s resolution, as per manual
+    }
+    if (!nh_private_.getParam("magnetic_field_stdev", magnetic_field_stdev_))
+    {
+        magnetic_field_stdev_ =
+            0.095 * (M_PI / 180.0);  // 0.095°/s as per manual
+    }
+    int data_interval_ms;
+    if (!nh_private.getParam("data_interval_ms", data_interval_ms))
+    {
+        data_interval_ms = 250;
```

Comment:
Let's keep the old default of 8 ms. A IMU publishing at 4 Hz is not useful in most use cases.

```suggestion
        data_interval_ms = 8;
```

Reply:
Yeah, agreed it is not that useful.  I was going what was listed as the default from the API: https://www.phidgets.com/?tier=3&catid=10&pcid=8&prodid=975#product_tab_5 .  But I can switch it back too.  I'll do the same for the individual accelerometer, magnetometer, and gyroscope as well.